lib/compat/wordpress-6.5/block-bindings/block-bindings.php

```diff
@@ -62,8 +68,96 @@ function wp_block_bindings_get_sources() {
  * @param string $source_value The value used to replace the HTML.
  * @return string The modified block content.
  */
-if ( ! function_exists( 'wp_block_bindings_replace_html' ) ) {
-	function wp_block_bindings_replace_html( $block_content, $block_name, $block_attr, $source_value ) {
-		return wp_block_bindings()->replace_html( $block_content, $block_name, $block_attr, $source_value );
+function gutenberg_block_bindings_replace_html( $block_content, $block_name, $block_attr, $source_value ) {
+			$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+	if ( null === $block_type ) {
+		return;
+	}
+
+	// Depending on the attribute source, the processing will be different.
+	switch ( $block_type->attributes[ $block_attr ]['source'] ) {
+		case 'html':
+		case 'rich-text':
+			$block_reader = new WP_HTML_Tag_Processor( $block_content );
+
+			// TODO: Support for CSS selectors whenever they are ready in the HTML API.
+			// In the meantime, support comma-separated selectors by exploding them into an array.
+			$selectors = explode( ',', $block_type->attributes[ $block_attr ]['selector'] );
+			// Add a bookmark to the first tag to be able to iterate over the selectors.
+			$block_reader->next_tag();
+			$block_reader->set_bookmark( 'iterate-selectors' );
+
+			// TODO: This shouldn't be needed when the `set_inner_html` function is ready.
+			// Store the parent tag and its attributes to be able to restore them later in the button.
+			// The button block has a wrapper while the paragraph and heading blocks don't.
```

Comment:
If you are backporting this file, make sure that multilines comments are with /* */

Reply:
Thanks Carlos! The backport is here btw https://github.com/WordPress/wordpress-develop/pull/5888 if you ever want to have a look.